source/server/connection_handler_impl.cc

```diff
@@ -288,5 +310,60 @@ ListenerStats ConnectionHandlerImpl::generateStats(Stats::Scope& scope) {
   return {ALL_LISTENER_STATS(POOL_COUNTER(scope), POOL_GAUGE(scope), POOL_HISTOGRAM(scope))};
 }
 
+ConnectionHandlerImpl::ActiveUdpListener::ActiveUdpListener(ConnectionHandlerImpl& parent,
+                                                            Network::ListenerConfig& config)
+    : ActiveUdpListener(parent, parent.dispatcher_.createUdpListener(config.socket(), *this),
+                        config) {}
+
+ConnectionHandlerImpl::ActiveUdpListener::ActiveUdpListener(ConnectionHandlerImpl& parent,
+                                                            Network::ListenerPtr&& listener,
+                                                            Network::ListenerConfig& config)
+    : ConnectionHandlerImpl::ActiveListenerBase(parent, std::move(listener), config),
+      udp_listener_(dynamic_cast<Network::UdpListener*>(listener_.get())) {
+
+  RELEASE_ASSERT(udp_listener_ != nullptr, "");
+
+  // Create the filter chain on creating a new udp listener
+  config_.filterChainFactory().createUdpListenerFilterChain(*this);
+
+  // If filter chain is empty, fail the creation of the listener
+  if (read_filters_.empty()) {
+    throw Network::CreateListenerException(fmt::format(
+        "Cannot create listener as no read filters registered for the udp listener: {} ",
+        config_.name()));
+  }
+}
+
+void ConnectionHandlerImpl::ActiveUdpListener::onData(Network::UdpRecvData& data) {
+  for (auto& filter : read_filters_) {
+    filter->onData(data);
+  }
+}
+
+void ConnectionHandlerImpl::ActiveUdpListener::onWriteReady(const Network::Socket&) {
+  // TODO(sumukhs): This is not used now. When write filters are implemented, this is a
```

Comment:
Then let's just delete all this code. We can add it back later when needed.

Reply:
Thing is, the OnWriteReady was already present in the UDP interface and there are tests exercising this code path. Are you saying we should remove all this and clean it up?

Reply:
Sorry, it's fine to leave if it's covered and we think it will be used later. Disregard.

Reply:
sounds good.